app/src/main/java/com/example/househomey/HomeFragment.java

```diff
@@ -131,6 +170,107 @@ private void showFilterMenu(View view) {
         });
 
         popupMenu.show();
+
+    }
+
+
+    // TODO: This is an interface method for a dialog fragment. Useless for now since I can't figure out how to have a dialog pop up inside a fragment
+    @Override
+    public void onOKPressed(){
+        for (int i=0;i<itemList.size();i++) {
+            if (itemList.get(i).isSelected()) {
+                itemRef
+                        .document(itemList.get(i).getId())
+                        .delete()
```

Comment:
We should probably look at finding a way to make this execute in a batch of some sorts. Like committing all the deletes at once. Cuz I think it's possible that after an item is deleted but before all items are deleted it could call the snapshot listener.

We could also just wrap them in a semaphore, but I think batching should be straightforward

Reply:
Yep got it. I can use WriteBatch to do this easily. Will make the change soon.